src/popover/internal.tsx

```diff
@@ -58,20 +58,29 @@ function InternalPopover(
 
   const [visible, setVisible] = useState(false);
 
+  const focusTrigger = useCallback(() => {
+    triggerRef.current?.focus();
+  }, []);
+
   const onTriggerClick = useCallback(() => {
     fireNonCancelableEvent(__onOpen);
     setVisible(true);
   }, [__onOpen]);
 
   const onDismiss = useCallback(() => {
     setVisible(false);
-  }, []);
+    focusTrigger();
+  }, [focusTrigger]);
 
-  const onTriggerKeyDown = useCallback((event: React.KeyboardEvent) => {
-    if (event.keyCode === KeyCode.tab || event.keyCode === KeyCode.escape) {
-      setVisible(false);
-    }
-  }, []);
+  const onTriggerKeyDown = useCallback(
+    (event: React.KeyboardEvent) => {
+      if (event.keyCode === KeyCode.tab || event.keyCode === KeyCode.escape) {
+        setVisible(false);
+        focusTrigger();
```

Comment:
I think the trigger should not be focused when received TAB keypress. The only exception is when expandToViewport is true -- in that case we can't focus elsewhere so focusing on the trigger is reasonable.

Reply:
good catch, this handler doesn't need to refocus ever in fact, as the trigger is by definition already focused...